includes/class-signature.php

```diff
@@ -356,8 +356,8 @@ public static function get_remote_key( $key_id ) {
 	public static function get_signature_algorithm( $signature_block ) {
 		if ( $signature_block['algorithm'] ) {
 			switch ( $signature_block['algorithm'] ) {
-				case 'rsa-sha-512':
-					return 'sha512'; // hs2019 https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-12.
+				case 'hs2019':
```

Comment:
```suggestion
				case 'hs2019':
				case 'rsa-sha-512':
```

let's keep it for backwards compatibility!

Reply:
I'm not sure it's compatible with anything. Is that string used anywhere?

Reply:
I have no idea, but I would fear so, because there was no error report since nearly two years. Aside from that: technically it is a major change if we remove it, because even if it might be a bug, we break backwards compatibility.

Reply:
it looks like the original implementation made some falsey assumptions. As @obenland said, I'm not sure any implementations declare `rsa-sha-512` as an algorithm.